msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.props

```diff
@@ -26,6 +26,20 @@ Copyright (C) 2013-2014 Xamarin. All rights reserved.
 		<_XamarinCommonPropsHasBeenImported>true</_XamarinCommonPropsHasBeenImported>
 	</PropertyGroup>
 
+
+	<!-- Story-time! MigrateToNewXMTFI is special because it un-does a lie we started telling from the 
+	beginning of Full (called Xamarin.Mac 4.5 back then). Back then, we released Modern (called Mobile) and 
+	when many customers upgraded from Classic they broke in terrible ways as none of their nugets worked. 
+	No nugets in the wild had heard of the Xamarin.Mac TFI and thus didn't support it. To "fix" this use case,
+	we created Full (XM 4.5) and _lied_ with our TFI claiming to be .NETFramework. Well we were close enough
+	for a vast majority of packages to work. However, this made creating Full specific package flavors impossible.
+	Now that we're getting "real" nuget support, we need to stop lying without breaking the world. We're starting by
+	allowing "opt-in" MigrateToNewXMTFI which will swap your declared TFI to the new hotness.
+	-->
+	<PropertyGroup Condition=" '$(MigrateToNewXMTFI)' == 'true' And '$(TargetFrameworkIdentifier)' != 'Xamarin.Mac' And '$(UseXamMacFullFramework)' == 'true' ">
```

Comment:
I don't really like abbreviations, so maybe name the variable something like: `UseXamarinMacTargetFrameworkIdentifier`?

Reply:
TFI (TargetFrameworkIdentifier) and TFV (TargetFrameworkVersion) are pretty standard msbuild jargon from what i can tell, but I concede the point. Let me think of a better name...

Reply:
`MigrateToNewXMIdentifier` ?

Reply:
`UseDesktopFrameworkIdentifier`?